### PR TITLE
Closes #30 Fix citizen spawns

### LINKED ISSUE #30: 1.8.9-0.2.113 - Citizen Spawns
1.8.9-0.2.113 - Playtest Feedback

Citizens can spawn in the ground, when the town hall chest is placed near a hill..  Had A citizen suffocate to death, and another take significant damage before they wandered out of the block.

### PR DESCRIPTION
**Pull request by cltnschlosser**
Increases y-direction search range, centers the entity, and add a small offset so they don't get stuck in the ground.

Review please @Minecolonies/maintainers

### CODE REVIEW

File: src/main/java/com/minecolonies/colony/Colony.java [hunk 1]
```diff
@@ -705,7 +704,7 @@ private void spawnCitizen(CitizenData data)
 
             entity.setColony(this, citizenData);
 
-            entity.setPosition(spawnPoint.getX(), spawnPoint.getY(), spawnPoint.getZ());
+            entity.setPosition(spawnPoint.getX() + 0.5D, spawnPoint.getY() + 0.1D, spawnPoint.getZ() + 0.5D);
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/colony/Colony.java]: MINOR Assign this magic number 0.5D to a well-named constant, and use the constant instead. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)
MINOR Assign this magic number 0.1D to a well-named constant, and use the constant instead. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)
MINOR Assign this magic number 0.5D to a well-named constant, and use the constant instead. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 3 issues
* MINOR 3 minor

Watch the comments in this conversation to review them.

**Comment by Kostronor** [comment]
:+1: Looks good @cltnschlosser do we want to fix the minor sonar things? I'm fine with it eigher way

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/60/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Comment by Raycoms** [comment]
:shipit:

**Pull request merged**